kitsune/search/v2/search.py

```diff
@@ -0,0 +1,238 @@
+import bleach
+from datetime import datetime
+from abc import ABC, abstractmethod
+
+from elasticsearch_dsl import Search as DSLSearch, Q
+from kitsune.search.v2.es7_utils import es7_client
+from kitsune.search.config import WIKI_DOCUMENT_INDEX_NAME, QUESTION_INDEX_NAME
+from kitsune.wiki.config import CANNED_RESPONSES_CATEGORY, TEMPLATES_CATEGORY
+
+HIGHLIGHT_TAG = "strong"
+
+
+def first_highlight(hit):
+    highlight = getattr(hit.meta, "highlight", None)
+    if highlight:
+        return bleach.clean(
+            # `highlight` is of type AttrDict, which is internal to elasticsearch_dsl
+            # when converted to a dict, it's like:
+            # `{ 'es_field_name' : ['highlight1', 'highlight2'], 'field2': ... }`
+            # so here we're getting the first item in the first value in that dict:
+            next(iter(highlight.to_dict().values()))[0],
```

Comment:
Elastic doesn't seem to have a way to specify "highlight any one of these fields" - just "highlight all these fields and only return one snippet for each" - however it can still fail to highlight a field at all, if nothing matches, so it's left to us to get the first matching snippet ourselves (with this monstrous line of code).